bearlyml23/inc/bearlyml23.h

```diff
@@ -0,0 +1,488 @@
+#define MIE_UEIE_MSK                  (1U << MIE_UEIE_POS)
+#define MIE_SEIE_POS                  0x09U
+#define MIE_SEIE_MSK                  (1U << MIE_SEIE_POS)
+#define MIE_VSEIE_POS                 0x0AU
+#define MIE_VSEIE_MSK                 (1U << MIE_VSEIE_POS)
+#define MIE_MEIE_POS                  0x0BU
+#define MIE_MEIE_MSK                  (1U << MIE_MEIE_POS)
+#define MIE_SGEIE_POS                 0x0CU
+#define MIE_SGEIE_MSK                 (1U << MIE_SGEIE_POS)
+
+#define MIE_USIP_POS                  0x00U
+#define MIE_USIP_MSK                  (1U << MIE_USIP_POS)
+#define MIP_SSIP_POS                  0x01U
+#define MIP_SSIP_MSK                  (1U << MIP_SSIP_POS)
+#define MIP_VSSIP_POS                 0x02U
+#define MIP_VSSIP_MSK                 (1U << MIP_VSSIP_POS)
+#define MIP_MSIP_POS                  0x03U
+#define MIP_MSIP_MSK                  (1U << MIP_MSIP_POS)
+#define MIE_UTIP_POS                  0x04U
+#define MIE_UTIP_MSK                  (1U << MIE_UTIP_POS)
+#define MIP_STIP_POS                  0x05U
+#define MIP_STIP_MSK                  (1U << MIP_STIP_POS)
+#define MIP_VSTIP_POS                 0x06U
+#define MIP_VSTIP_MSK                 (1U << MIP_VSTIP_POS)
+#define MIP_MTIP_POS                  0x07U
+#define MIP_MTIP_MSK                  (1U << MIP_MTIP_POS)
+#define MIP_SEIP_POS                  0x09U
+#define MIP_SEIP_MSK                  (1U << MIP_SEIP_POS)
+#define MIP_VSEIP_POS                 0x0AU
+#define MIP_VSEIP_MSK                 (1U << MIP_VSEIP_POS)
+#define MIP_MEIP_POS                  0x0BU
+#define MIP_MEIP_MSK                  (1U << MIP_MEIP_POS)
+#define MIP_SGEIP_POS                 0x0CU
+#define MIP_SGEIP_MSK                 (1U << MIP_SGEIP_POS)
+
+
+/* ================================ Peripheral Definitions ================================ */
+
+/* Peripheral Struct Definition */
+typedef struct {
+  __IO uint32_t MSIP0;                          /** MSIP Registers (1 bit wide) */
+  __IO uint32_t MSIP1;                          /** MSIP Registers (1 bit wide) */
+  __IO uint32_t MSIP2;                          /** MSIP Registers (1 bit wide) */
+  __IO uint32_t MSIP3;                          /** MSIP Registers (1 bit wide) */
+  uint32_t RESERVED0[4091];
+  __IO uint64_t MTIMECMP0;                      /** MTIMECMP Registers */
+  uint32_t RESERVED1[8188];
+  __IO uint64_t MTIME;                          /** Timer Register */
+} CLINT_TypeDef;
+
+
+typedef struct {
+  __IO uint32_t priority_threshold;
+  __IO uint32_t claim_complete;
+} PLIC_ContextControl_TypeDef;
+
+typedef struct {
+  __IO uint32_t priorities[1024];
+  __I  uint32_t pendings[1024];
+  __IO uint32_t enables[1024];
+} PLIC_TypeDef;
+
+// because the maximum struct size is 65535, we need to split PLIC content
+typedef struct {
+  PLIC_ContextControl_TypeDef context_controls[1024];
+} PLIC_Extra_TypeDef;
+
+
+typedef struct {
+  __IO uint32_t TILE0_RESET;      /** RocketTile 0 */   // 0x00
```

Comment:
Make sure the RCC (Reset and Clock Control) are correct. Do we still have reset and clock gate for all cores? How many cores are there?